### PR TITLE
Adding missing variable to be able to run the anti-electron discriminator on present trainings

### PR DESCRIPTION
**Pull request by roger-wolf**
Dear all, 

as discussed in the RECO/AT meeting last week this is a PR to add one more missing variable (float phiAtEcalEntrance_) to the TauPFEssentials.h. Usual tests have been made. Neither runtime performance nor event size are compromised as expected. We have checked that the variable is indeed written to file. The modification has been communicated to the miniAOD officers and those have notified the PPD. 

Cheers,
Roger

### CODE REVIEW

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by roger-wolf** [comment]
Dear colleagues, 

I don't want to be impatient, but I have a second PR in the queue now that relies on this one to be integrated first. I suppose that this one can be signed off very quickly. Alternatively I can of course just make the new PR and indicate that it supersedes this one. The only issue I have with it is that I want this one to be integrated in any case into 81X before the deadline even if the more extended one (still to come) would still be in review. 

Bottom line: if you could sign this one off this ease my life a lot. E.g. I could have saved all the arguments above :-)... 

Cheers,
Roger

**Comment by cvuosalo** [comment]
urgent

**Comment by cvuosalo** [comment]
+1

For #15995 29461219cfa3464839ce0f62dac53a4e1ff9639a

For taus, adding a missing variable for the anti-electron discriminator.

The code changes are satisfactory, and Jenkins tests against baseline CMSSW_8_1_X_2016-09-27-1100 show no significant differences, as expected. A test of workflow 25202.0 with 20 events against baseline CMSSW_8_1_0_pre12 also shows no significant differences. Compatibility is preserved so the new code can still read the previous Mini-AOD version. The new variable increases the size of Mini-AOD very slightly:
```
-----------------------------------------------------------------
   or, B         new, B      delta, B   delta, %   deltaJ, %    branch 
-----------------------------------------------------------------
   5303.7 ->      5310.9          7      0.1   0.00     patTaus_slimmedTausBoosted__RECO.
   5760.3 ->      5774.9         15      0.3   0.01     patTaus_slimmedTaus__RECO.
-------------------------------------------------------------
   178534 ->      178558         24             0.0     ALL BRANCHES
```
